src/lib.rs

```diff
@@ -693,7 +695,7 @@ impl Arguments {
         let mut flags_left = Vec::new();
         for arg in &self.0 {
             if let Some(s) = arg.to_str() {
-                if s.starts_with('-') && s != "-" {
+                if s.starts_with('-') && s != "-" && !s.chars().nth(1).unwrap_or('f').is_numeric() {
```

Comment:
You can replace this with `s.chars().skip(1).all(|c| c.is_ascii_digit() || c == '.'))`